shiny/workflowPlot/bm.R

```diff
@@ -0,0 +1,275 @@
+})
+
+observeEvent({
+  bm$ready
+  bm$metrics_list
+  bm$plots_list
+  bm$vars_list
+},{
+  
+  plot_ind <- grep("_plot",bm$metrics$name)
+  
+  output$bm_inputs <- renderUI({
+    if(bm$ready){
+      list(
+        column(4, wellPanel(
+          checkboxGroupInput("vars", label = h3("Variables"),
+                             choiceNames = bm$vars$pecan_name,
+                             choiceValues = bm$vars$variable_id),
+          # actionButton("selectall.var","Select /Deselect all variables"),
+          label=h6("Label")
+        )),
+        column(4, wellPanel(
+          checkboxGroupInput("metrics", label = h3("Numerical Metrics"), 
+                             choiceNames = bm$metrics$description[-plot_ind],
+                             choiceValues = bm$metrics$id[-plot_ind]),
+          # actionButton("selectall.num","Select/Deselect all numerical metrics") ,
+          label=h6("Label")
+        )),
+        column(4, wellPanel(
+          checkboxGroupInput("plots", label = h3("Plot Metrics"),
+                             choiceNames = bm$metrics$description[plot_ind],
+                             choiceValues = bm$metrics$id[plot_ind]),
+          # actionButton("selectall.plot","Select/Deselect all plot metrics"),
+          label=h6("Label")
+        ))
+      )
+    }
+  })
+  if(bm$ready){bm$calc_bm_message <- sprintf("Please select at least one variable and one metric")}
+})
+
+observeEvent({
+  input$vars
+  input$metrics
+  input$plots
+  },{
+  v <- ifelse(is.null(input$vars),0,length(input$vars))
+  n <- ifelse(is.null(input$metrics),0,length(input$metrics))
+  p <- ifelse(is.null(input$plots),0,length(input$plots))
+  m <- n + p
+  output$report <- renderText(sprintf("Number of vars: %0.f, Number of metrics: %0.f", v,m))
+  if(v > 0 & m > 0){
+    output$calc_bm_button <- renderUI({actionButton("calc_bm", "Calculate Benchmarks")})
+    bm$bm_vars <- input$vars
+    bm$bm_metrics <- c()
+    if(n > 0) bm$bm_metrics <- c(bm$bm_metrics, input$metrics)
+    if(p > 0) bm$bm_metrics <- c(bm$bm_metrics, input$plots)
+  }
+  
+}, ignoreNULL = FALSE)
+
+observeEvent(input$calc_bm,{
+  req(input$all_input_id)
+  req(input$all_site_id)
+  req(bm)
+  bm$calc_bm_message <- sprintf("Setting up benchmarks")
+  output$reportvars <- renderText(paste(bm$bm_vars, seq_along(bm$bm_vars)))
+  output$reportmetrics <- renderText(paste(bm$bm_metrics))
+  
+  inputs_df <- getInputs(bety,c(input$all_site_id)) %>% 
+    dplyr::filter(input_selection_list == input$all_input_id)
+  output$inputs_df_table <- renderTable(inputs_df)
+  
+  config.list <- PEcAn.utils::read_web_config("../../web/config.php")
+  output$config_list_table <- renderTable(as.data.frame.list(config.list))
+  
+  bm$bm_settings$info <- list(userid = 1000000003) # This is my user id. I have no idea how to get people to log in to their accounts through the web interface and right now the benchmarking code has sections dependent on user id - I will fix this. 
```

Comment:
wasn't userid among the things that you deleted back on L25-30? Seems like you should save that for here. Though I guess that's not a perfect solution because that's the userid of the person that did the run, not the person that set it up as a benchmark.